pycarol/storage.py

```diff
@@ -147,6 +153,7 @@ def save(
 
 
         """
+        self._init_if_needed()
```

Comment:
for all the `_init_if_needed`

If you see `StorageGCPCS` (that is the backend) always calls `carolina.get_client()` and this is always checking if it needs to update tokens. 

This is adding a overhead I think